### PR TITLE
Fix crony.d config directory in Ansible in rule chronyd_or_ntpd_set_maxpoll

### PR DESCRIPTION
**Pull request by jan-cerny**
The Ansible remediation manipulates unrelated files in /etc because the support for chrony.d configuration directory was implemented wrong. This patch reworks the support for chrony.d configuration directory.

For more details, please read commit messages of all commits.

### CODE REVIEW

**Comment by jan-cerny** [comment]
This PR has been originally intended as a fix for https://github.com/ComplianceAsCode/content/issues/11934. However, it turned out that only the order of the rules is the cause of the reported issue. To fix the issue, it isn't necessary to fix the  the support for chrony.d configuration directory. To fix the issue we only need to reorder the rules. As this PR needs a review from broader community, the ordering change has been extracted out to a new separate PR: https://github.com/ComplianceAsCode/content/pull/11960

**Review by marcusburghardt — APPROVED** [review]
LGTM. Thanks

**Comment by marcusburghardt** [comment]
FYI @ComplianceAsCode/ubuntu-maintainers @ComplianceAsCode/suse-maintainers @ComplianceAsCode/oracle-maintainers

File: linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_d_one_server_misconfigured.fail.sh
Comment (mpurg) [inline linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_d_one_server_misconfigured.fail.sh:9]: These configs seem platform specific. Maybe the tests should be made platform specific?

File: linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_d_one_pool_misconfigured.fail.sh
Comment (mpurg) [inline linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_d_one_pool_misconfigured.fail.sh:7]: These configs seem platform specific. Maybe the tests should be made platform specific?

**Review by mpurg — COMMENTED** [review]
This could be a good time to fix the hardcoded (and thus broken) paths in OVAL and bash - `/etc/chrony.conf`, `/etc/chrony.d/*conf`, though it could be done in a followup PR as well.

**Comment by mpurg** [comment]
Slightly unrelated, but after looking at the OVAL, it seems that it passes when 'server/poll' directives are missing in the config files (or the config files are not present at those hardcoded paths). According to the Ubuntu/RHEL STIGs, the directives should be explicitly defined (and not in ntp configs).

**Comment by jan-cerny** [comment]
I have updated in the rule chronyd_or_ntpd_set_maxpoll:

- consistent support for the configuration directory
- always use product properties `chrony_conf_path` and `chrony_d_path`
  for the configuration file and configuration directory instead of
  using hard-coded path
- do not pass if no server is set, this align the behavior with RHEL
  and Ubuntu STIGs

File: linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_no_pool_nor_servers.fail.sh
Comment (Xeicker) [inline linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_no_pool_nor_servers.fail.sh:4]: Now that you are modifying this file, I think this test is not necessarily just for stig profile

Comment (jan-cerny) [reply linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_no_pool_nor_servers.fail.sh:4]: Good point! Also, we should move away from using the `profiles` tag if the tests aren't regression tests valid only in context of the profile, for parametrize we should use `variables` keyword instead. I'll try to do something with that.

**Comment by jan-cerny** [comment]
I have changed the test scenarios to use variables instead of profiles key.

File: linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_d_one_pool_misconfigured.fail.sh
Comment (teacup-on-rockingchair) [reply linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/tests/chrony_d_one_pool_misconfigured.fail.sh:8]: I tend to agree with the above comment, instead of making it platform specific we can replace `20-pools.conf` with wildcard

**Comment by jan-cerny** [comment]
@teacup-on-rockingchair PTAL

**Comment by teacup-on-rockingchair** [comment]
> @teacup-on-rockingchair PTAL

I guess you missed @mpurg and mine comments on the tests above? https://github.com/ComplianceAsCode/content/pull/11958#discussion_r1607576043

**Comment by jan-cerny** [comment]
I have marked platform specific scenarios.

**Review by teacup-on-rockingchair — APPROVED** [review]
Nice one :bow:

**Comment by marcusburghardt** [comment]
@Xeicker , could you also review, please? It also needs approval from @ComplianceAsCode/oracle-maintainers .

**Review by marcusburghardt — APPROVED** [review]
Also reviewed the updates after my first review. Thanks @jan-cerny

**Review by Xeicker — APPROVED** [review]

**Pull request merged**
